bpfd/src/lib.rs

```diff
@@ -59,12 +76,13 @@ pub async fn serve(config: Config, static_programs: Vec<StaticPrograms>) -> anyh
     let serve = Server::builder()
         .tls_config(tls_config)?
         .add_service(LoaderServer::with_interceptor(loader, intercept))
-        .serve(addr);
+        .serve_with_shutdown(addr, shutdown_rx.map(drop));
 
     tokio::spawn(async move {
         info!("Listening on {addr}");
         if let Err(e) = serve.await {
             eprintln!("Error = {e:?}");
+            signal_handler.abort();
```

Comment:
This is only going to terminate the the listener though, not the larger loop below that receives messages from the channel. We don't want to panic here as it will prevent anything else from gracefully cleaning up, but the error should be logged.